drivers/virtualbox/virtualbox.go

```diff
@@ -288,26 +289,37 @@ func (d *Driver) Create() error {
 		return err
 	}
 
-	// let VBoxService do nice magic automounting (when it's used)
-	if err := vbm("guestproperty", "set", d.MachineName, "/VirtualBox/GuestAdd/SharedFolders/MountPrefix", "/"); err != nil {
-		return err
-	}
-	if err := vbm("guestproperty", "set", d.MachineName, "/VirtualBox/GuestAdd/SharedFolders/MountDir", "/"); err != nil {
-		return err
-	}
+	var shareMountDir, shareDir, username string
 
-	var shareName, shareDir string // TODO configurable at some point
 	switch runtime.GOOS {
 	case "windows":
-		shareName = "c/Users"
-		shareDir = "c:\\Users"
+		username = os.Getenv("USERNAME")
+		shareDir = "c:\\Users\\" + username
+		shareMountDir = "/c/Users"
 	case "darwin":
-		shareName = "Users"
-		shareDir = "/Users"
-		// TODO "linux"
+		username = os.Getenv("USER")
+		shareDir = path.Join("/Users", username)
+		shareMountDir = "/Users"
+	case "linux":
+		username = os.Getenv("USER")
+		shareMountDir = "/home"
+		shareDir = path.Join("/home", username)
+	}
+
+	shareName := username
+
+	log.Debugf("creating share: path=%s", shareDir)
+
+	// let VBoxService do nice magic automounting (when it's used)
+	if err := vbm("guestproperty", "set", d.MachineName, "/VirtualBox/GuestAdd/SharedFolders/MountDir", shareMountDir); err != nil {
```

Comment:
The point of setting this to `/` is that VirtualBox will then take a name like `home/tianon` and mount it at `/home/tianon` automatically.  Setting this to `/home` means that any automount shares added after the machine is created will live under `/home` (so a name of `mnt/files` would automount as `/home/mnt/files` instead of as `/mnt/files`).